### DISCUSSION TITLE
Some mind maps about manim structure and process

### DISCUSSION THREAD

**Discussion by TonyCrane** [3b1b/manim#1317] [Show and tell]
Hello everyone.

I made some mind maps of the structure and operation process of manim for both the master and the shaders branch. And there are both Chinese and English versions.
You can reprint and use it at will, but you need to indicate the original author @TonyCrane (鹤翔万里 in Chinese). That is to follow the CC-BY 4.0 license.

At present, the shaders branch is not yet complete, so the mind maps about the shaders branch will be updated from time to time.

Hope they can help you in some ways. :D

- Structure of manim in the master branch
![manim_master_structure](https://user-images.githubusercontent.com/44120331/104811848-59ecbf80-5839-11eb-9f14-7af8caa31d92.png)
[manim_master_structure.pdf](https://github.com/3b1b/manim/files/5824383/manim_shaders_structure.pdf)

- Structure of manim in the cairo-backend branch
![manim_backend_structure](https://user-images.githubusercontent.com/44120331/104811827-36297980-5839-11eb-846a-bd275b9ac530.png)
[manim_backend_structure.pdf](https://github.com/3b1b/manim/files/5824384/manim_master_structure.pdf)

- Process of manim in the master branch
![manim_master_process_en](https://user-images.githubusercontent.com/44120331/104812044-7806ef80-583a-11eb-91e1-5eebcd7e173f.png)
[manim_master_process_en.pdf](https://github.com/3b1b/manim/files/5824373/manim_shaders_process_en.pdf)
[manim_master_process_cn.pdf](https://github.com/3b1b/manim/files/5824374/manim_shaders_process_cn.pdf)

- Process of manim in the cairo-backend branch
![manim_cairo-backend_process_en](https://user-images.githubusercontent.com/44120331/104812032-658cb600-583a-11eb-9bbd-9d5eb2ed20a0.png)
[manim_cairo-backend_process_en.pdf](https://github.com/3b1b/manim/files/5824379/manim_master_process_en.pdf)
[manim_cairo-backend_process_cn.pdf](https://github.com/3b1b/manim/files/5824380/manim_master_process_cn.pdf)

**Comment by Splines** [comment]
This is gold, thanks a lot ❤

Just wanted to note that I've extracted some parts of the `main()` entry point to a new `ReloadManager`:

```py
class ReloadManager:
"""
Manages the loading and running of scenes and is called directly from the
main entry point of ManimGL.

The name "reload" comes from the fact that this class handles the
reinitialization of scenes when requested by the user via the `reload()`
command in the IPython shell.
"""
```

This is already in the `master` branch via #2240 where I introduced a new `reload()` command. It will be enhanced via #2257 and an auto-reload option is offered via #2252.
